### ISSUE TITLE
How to render map roads properly?

### ISSUE THREAD

**Issue by interpegasus** [mapbox/mapbox-unity-sdk#1715]
Hi I am new with Unity and Mapbox. And I would like to know **how to render the main roads** in a city, including applying a texture?

Currently I am facing two issues:

1. When selecting the road data layer the map displays too many roads. I would like to display only one big road.
2. There is z-fighting on intersections. I tried the NoiseOffsetModifer but there is flickering when playing the scene.

Thank you in advance!

<img width="391" alt="Screen Shot 2020-11-04 at 7 32 04 PM" src="https://user-images.githubusercontent.com/934795/98183608-82217f00-1ed6-11eb-8fd4-373eab7d2ae5.png">
<img width="347" alt="Screen Shot 2020-11-04 at 7 31 38 PM" src="https://user-images.githubusercontent.com/934795/98183618-86e63300-1ed6-11eb-8dee-7aba04de7fc5.png">

**Configuration**
* Unity version: 2019.2.9f1
  * Scripting Runtime Version: default Scripting Runtime Version .NET 4.6
  * Scripting Backend: Default
  * Api Compatibility Level: .NET Standard 2.0
* Mapbox SDK version: 2.1.1
* The platform you're building to: WebGL / iOS / Android

**Comment by janweigel** [comment]
Same problem here.
Did you manage to fix it?
I tried everything with modifiers and settings but non is working so far.

**Comment by interpegasus** [comment]
Related Issue:
https://github.com/mapbox/mapbox-unity-sdk/issues/1606

**Comment by interpegasus** [comment]
Hi @janweigel It looks like this is perhaps a bug on the SDK, since it is not possible to filter out non-primary roads.
While a fix is implemented, one solution is selecting the **_road_label_** data layer instead of the **_road_** data layer. In that way it only displays one road instead of too many.

**Comment by janweigel** [comment]
Hi @interpegasus, thanks for your feedback.
Unfortunately the layer road_label doesn't even exist in Mapbox Unity SDK 2.1.1 therefore I'm not bale to give it a try.
Could you give me some more information how to get the road_label layer, please.
Thanks!

**Comment by interpegasus** [comment]
Hi @janweigel It can be found via the map features section in Unity Editor:
Features -> Map Features -> Add Feature -> Roads
Data Layer -> road_label

The data source is **Mapbox Streets V7.** 
I hope it helps!
<img width="467" alt="Screen Shot 2020-11-12 at 1 12 50 PM" src="https://user-images.githubusercontent.com/934795/98979059-dd5ffc80-24e8-11eb-82fa-1ebbd6e7b9ce.png">
<img width="695" alt="Screen Shot 2020-11-12 at 1 17 27 PM" src="https://user-images.githubusercontent.com/934795/98979532-77c04000-24e9-11eb-954d-653483939b45.png">

**Comment by janweigel** [comment]
@interpegasus that's strange. The Data Layer road_label doesn't exist in my version.
Which Data Source are you using?
Thanks for the detailed information!

**Comment by brnkhy** [comment]
@interpegasus @janweigel Unfortunately creating good looking roads are rather difficult as there is no information on road width or lane count. Intersections aren't always marked in current vector data either making it even worse.

@janweigel according to [docs](https://docs.mapbox.com/vector-tiles/reference/mapbox-streets-v8/), `road_label` layer is merged into `road` layer in v8 so you are probably using that?
Noise offset modifier is the simplest thing to avoid z-fights but it just randomly moves things so I guess there's no guarantees things won't overlap afterwards.

**Comment by janweigel** [comment]
@brnkhy yes, I'm using V8, so that's probably the cause of the missing road_label layer.
I already tried the noise modifier but it causes heavy flickering when the map/camera is rotated.
I wonder how other games (e.g. the Niantic ones) manage to combine all the roads to one polygon and not to many overlapping ones. At least it seems like that when I take a look at how the textures at intersections.
Maybe a much more complex modifier to convert the lines to a polygon would be necessary.
Until now I didn't any solution to get acceptable looking textured roads.
